community/cypher/cypher-logical-plans-3.4/src/main/scala/org/neo4j/cypher/internal/v3_4/logical/plans/LogicalPlan.scala

```diff
@@ -71,6 +71,9 @@ abstract class LogicalPlan
 
   override def rememberMe(old: AnyRef): Unit = _id = old.asInstanceOf[LogicalPlan]._id
 
+  // This should only be used for converting old versions of LogicalPlans
+  def setIdTo(id: LogicalPlanId) : Unit = _id = Some(id)
```

Comment:
The logical plan abstraction is leaky... we might want to revisit that later.

Reply:
Yeah, it's kinda sucky, I agree